be/src/olap/memtable_flush_executor.cpp

```diff
@@ -0,0 +1,177 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "olap/memtable_flush_executor.h"
+
+#include "olap/data_dir.h"
+#include "olap/delta_writer.h"
+#include "olap/memtable.h"
+#include "runtime/exec_env.h"
+
+namespace doris {
+
+OLAPStatus FlushHandler::submit(std::shared_ptr<MemTable> memtable) {
+    RETURN_NOT_OK(_last_flush_status.load());
+    MemTableFlushContext ctx;
+    ctx.memtable = memtable;
+    ctx.flush_handler = this->shared_from_this();
+    _flush_futures.push(_flush_executor->_push_memtable(_flush_queue_idx, ctx));
```

Comment:
One of future and callback is enough. No need to have both, which will make it more complex.
In this case, what you need is CountDownLatch, update in callback.